api/v1alpha1/openapi.yaml

```diff
@@ -2335,17 +2389,21 @@ components:
         observedGeneration:
           type: integer
           format: int64
+          minimum: 0
           description: The .metadata.generation that the condition was set based upon.
-        lastHeartbeatTime:
-          type: string
         lastTransitionTime:
           type: string
+          format: date-time
           description: The last time the condition transitioned from one status to another.
         message:
           type: string
+          maxLength: 32768
           description: Human readable message indicating details about last transition.
         reason:
           type: string
+          minLength: 1
+          maxLength: 1024
+          pattern: '^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$'
```

Comment:
Do we need to be concerned with outputs?  Same for the other validations for Conditions.

Reply:
I've dropped validation of Conditions for now, but think that anything that can be modified via the API needs validation. Will add in a follow-up PR.